samples/sensor/hts221/sample.yaml

```diff
@@ -1,7 +1,18 @@
 sample:
   name: HTS221 Temperature and Humidity Monitor
+common:
+    tags: sensor
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: false
+      regex:
+        - "Temperature:(.*)C"
+        - "Relative Humidity:(.*)%"
+      fixture: fixture_i2c_hts221 # see documentation
 tests:
   test:
     harness: sensor
     tags: samples sensor
-    depends_on: i2c hts221
+    depends_on: i2c
+    platform_whitelist: quark_d2000_crb
```

Comment:
why is this being whitelisted on quark_d2000? Having to remove the depend_on on hts221 is going to cause problems, I know why this was added and was a workaround, but it seems we have a conflict between the fixture and depends_on.

Reply:
> why is this being whitelisted on quark_d2000?

Sanitycheck reports build failure for this test(samples/sensor/hts221) on multiple boards that have i2c capability, but does not implement hts221 sensor driver. I have validated it on quark_d2000 that is connected in our automation. So, whitelisting only on particular platform that passes the test to avoid sanity check failure.

**Sanity check failure Log:**
```
[ 89%] Building C object zephyr/drivers/sensor/hts221/CMakeFiles/drivers__sensor__hts221.dir/hts221.c.obj
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/sensor/hts221/hts221.c: In function ‘hts221_init’:
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/drivers/sensor/hts221/hts221.c:110:37: error: ‘CONFIG_HTS221_I2C_MASTER_DEV_NAME’ undeclared (first use in this function)
  drv_data->i2c = device_get_binding(CONFIG_HTS221_I2C_MASTER_DEV_NAME);
                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
**Shippable link:** https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/22154/3/tests
The following boards report this error: frdm_kl25z hexiwear_k64 frdm_k64f olimexino_stm32 nucleo_f070rb stm32f072b_disco stm32f3_disco frdm_kw41z reel_board nucleo_l053r8 96b_carbon stm32f723e_disco nucleo_f429zi

@inakypg: Can you please suggest a better way instead for hard-coding for a particular platform as platform_whitelist. Possibly, one way is to enable hts221 dependency only for the boards that support this sensor driver. Currently, hts221 is just a dummy value here as rightly mentioned by @nashif.